### ISSUE TITLE
PrepareIotSample.ps1 error s with Azure cmdlets 1.3.0

### ISSUE THREAD

**Issue by robomotic** [Azure/azure-iot-remote-monitoring#134]
Hello there,
I am attempting a local deploy with the latest master code + VS2015 (Enterprise) + Azure Cmdlets 1.3.0 but I am getting several errors as below.
Any idea about how to fix this?

```
C:\Users\ddiprod\Documents\DashboardTest\azure-iot-remote-monitoring-master>C:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe -ExecutionPolicy Unrestricted -Command C:\Users\ddiprod\Documents\DashboardTest\azure-iot-remote-monitoring-master\\Common\Deployment\PrepareIoTSample.ps1 -environmentName local -configuration Debug
WARNING: This script Azure Cmdlets was tested with 1.0.3
WARNING: Found 1.3.0 installed; continuing, but errors might
occur
Import-Module : The required module 'Azure.Storage' is not
loaded. Load the module or remove the module from
'RequiredModules' in the file 'C:\Program Files (x86)\Microsoft
SDKs\Azure\PowerShell\ServiceManagement\Azure\Azure.psd1'.
At C:\Users\ddiprod\Documents\LingangDashboard\azure-iot-remote-m
onitoring-master\Common\Deployment\DeploymentLib.ps1:913 char:37
+         Get-ChildItem $modulePath | Import-Module
+                                     ~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (C:\Program F
   ile...zure\Azure.psd1:String) [Import-Module], MissingMember
  Exception
    + FullyQualifiedErrorId : Modules_InvalidManifest,Microsoft.
   PowerShell.Commands.ImportModuleCommand

nuget.exe not found. Downloading from http://www.nuget.org/nuget.exe ...
Library Microsoft.IdentityModel.Clients.ActiveDirectory not found, loading...
Microsoft.IdentityModel.Clients.ActiveDirectory Library Nuget doesn't exist. Downloading now ...
Loading Microsoft.IdentityModel.Clients.ActiveDirectory.dll Assembly ...
Library WindowsAzure.ServiceBus not found, loading...
WindowsAzure.ServiceBus Library Nuget doesn't exist. Downloading now ...
Loading Microsoft.ServiceBus.dll Assembly ...
Library WindowsAzure.Storage not found, loading...
WindowsAzure.Storage Library Nuget doesn't exist. Downloading now ...
Loading Microsoft.WindowsAzure.Storage.dll Assembly ...
Get-AzureAccount : The 'Get-AzureAccount' command was found in
the module 'Azure', but the module could not be loaded. For more
information, run 'Import-Module Azure'.
At C:\Users\ddiprod\Documents\LingangDashboard\azure-iot-remote-m
onitoring-master\Common\Deployment\DeploymentLib.ps1:537 char:16
+     $account = Get-AzureAccount
+                ~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Get-AzureAccount:
   String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CouldNotAutoloadMatchingModule

Signing you into Azure...
Add-AzureAccount : The 'Add-AzureAccount' command was found in
the module 'Azure', but the module could not be loaded. For more
information, run 'Import-Module Azure'.
At C:\Users\ddiprod\Documents\LingangDashboard\azure-iot-remote-m
onitoring-master\Common\Deployment\DeploymentLib.ps1:542 char:20
+         $account = Add-AzureAccount
+                    ~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Add-AzureAccount:
   String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CouldNotAutoloadMatchingModule

Logging in
Login-AzureRmAccount : The term 'Login-AzureRmAccount' is not
recognized as the name of a cmdlet, function, script file, or
operable program. Check the spelling of the name, or if a path
was included, verify that the path is correct and try again.
At C:\Users\ddiprod\Documents\LingangDashboard\azure-iot-remote-m
onitoring-master\Common\Deployment\DeploymentLib.ps1:559 char:9
+         Login-AzureRmAccount | Out-Null
+         ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Login-AzureRmAcco
   unt:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Save-AzureRmProfile : The term 'Save-AzureRmProfile' is not
recognized as the name of a cmdlet, function, script file, or
operable program. Check the spelling of the name, or if a path
was included, verify that the path is correct and try again.
At C:\Users\ddiprod\Documents\LingangDashboard\azure-iot-remote-m
onitoring-master\Common\Deployment\DeploymentLib.ps1:560 char:9
+         Save-AzureRmProfile -Path $profilePath
+         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Save-AzureRmProfi
   le:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Get-AzureRmSubscription : The term 'Get-AzureRmSubscription' is
not recognized as the name of a cmdlet, function, script file,
or operable program. Check the spelling of the name, or if a
path was included, verify that the path is correct and try again.
At C:\Users\ddiprod\Documents\LingangDashboard\azure-iot-remote-m
onitoring-master\Common\Deployment\DeploymentLib.ps1:790 char:21
+         $accounts = Get-AzureRmSubscription
+                     ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Get-AzureRmSubscr
   iption:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Available subscriptions:
You cannot call a method on a null-valued expression.
At C:\Users\ddiprod\Documents\LingangDashboard\azure-iot-remote-m
onitoring-master\Common\Deployment\DeploymentLib.ps1:800 char:20
+ ...      while
(!$accounts.SubscriptionId.Contains($global:SubscriptionId ...
+
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeE
   xception
    + FullyQualifiedErrorId : InvokeMethodOnNull

Select-AzureRmSubscription : The term
'Select-AzureRmSubscription' is not recognized as the name of a
cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the
path is correct and try again.
At C:\Users\ddiprod\Documents\LingangDashboard\azure-iot-remote-m
onitoring-master\Common\Deployment\DeploymentLib.ps1:823 char:5
+     Select-AzureRmSubscription -SubscriptionId
$global:SubscriptionId
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Select-AzureRmSub
   scription:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Available locations:

Option Region
------ ------
     1 East US
     2 North Europe
     3 East Asia
     4 West US
     5 West Europe
     6 Southeast Asia



Select an option from the above list:
```

**Comment by kylecaulfield** [comment]
I have the same issue if I don't run the dev command prompt for vs as administrator; have you tried that?

**Comment by jamesweb-ms** [comment]
I have typically seen this error:

Import-Module : The required module 'Azure.Storage' is not
loaded. Load the module or remove the module from
'RequiredModules' in the file 'C:\Program Files (x86)\Microsoft
SDKs\Azure\PowerShell\ServiceManagement\Azure\Azure.psd1'.

 if you install powershell and then try to use it right away.  A reboot is actually required... though it doesn't tell you.

**Comment by aguilaaj** [comment]
As called out in #144: Make sure you have the correct prerequisites outlined here: https://github.com/Azure/azure-iot-remote-monitoring/wiki/Set-up-development-environment-%28Windows%29

**Comment by robomotic** [comment]
Yes sorry forgot to mention but restarting the laptop worked pretty well and was able to completed the build.

**Comment by robomotic** [comment]
Oh no actually there was an additional error I incurred into, when I rerun the local deploy.

Signed into Azure already
Logging in
Available subscriptions:

Option SubscriptionName SubscriptionId
------ ---------------- --------------
     1 BizSpark         ******-******-4088-9b30-56df9f70671d


Set-AzureRmContext : Cannot validate argument on parameter 'SubscriptionId'.
The argument is null or empty. Provide an argument that is not null or empty,
and then try the command again.
At C:\Users\ddiprod\Documents\LingangDashboard\azure-iot-remote-monitoring-mast
er\Common\Deployment\DeploymentLib.ps1:823 char:48
+     Select-AzureRmSubscription -SubscriptionId $global:SubscriptionId
+                                                ~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Set-AzureRmContext], Parameter
   BindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Microsoft.Azure
   .Commands.Profile.SetAzureRMContextCommand

I can then select the locations but I fear that this error is what gives more issue when I actually try to run the solution.

**Comment by aguilaaj** [comment]
@jamesweb-ms has been taking a look for the fix. 
For now you should try this workaround to clean-up your account: 
Get-AzureAccount | %{Remove-AzureAccount -Name $_.Id -Force}

**Comment by robomotic** [comment]
Okay, so I did that from a cmd.exe running as administator .
I start powershell and then do:

>  Get-AzureAccount | %{Remove-AzureAccount -Name $_.Id -Force}
> WARNING: The default subscription is being removed. Use
> Select-AzureSubscription -Default <subscriptionName> to select a new default
> subscription.
Then I exit from powershell prompt.
The I re-run the build.cmd local command but I am still getting the same error:

> Set-AzureRmContext : Cannot validate argument on parameter 'SubscriptionId'.

**Comment by aguilaaj** [comment]
You can also try to delete the user file in the root folder of where you pulled down the repository. It will be named something like <username>.user then retry.

**Comment by robomotic** [comment]
I did that as well but then when I re run the build.cmd local I still get the same error.

**Comment by robomotic** [comment]
Okay so I have started afresh, deleted all configuration files, commandline with admin priviege, with the correct azure powershell version, I get now an additional error before the original one:
```

C:\Users\pippo\Documents\LingangDashboard\azure-iot-remote-monitoring-master\
Common\Deployment\DeploymentLib.ps1?
[D] Do not run  [R] Run once  [S] Suspend  [?] Help (default is "D"): R
Exception calling "CompareTo" with "1" argument(s): "Object must be of type
Version."
At C:\Users\pippo\Documents\LingangDashboard\azure-iot-remote-monitoring-mast
er\Common\Deployment\DeploymentLib.ps1:884 char:1
+ $comparison = $expected.CompareTo($module.Version)
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentException

Library WindowsAzure.ServiceBus not found, loading...
Loading Microsoft.ServiceBus.dll Assembly ...
Signing you into Azure...
Logging in
Available subscriptions:

Option SubscriptionName SubscriptionId
------ ---------------- --------------
     1 BizSpark         f47ad81e-xxxx-xxxx-xxxx-56df9f70671d


Set-AzureRmContext : Cannot validate argument on parameter 'SubscriptionId'.
The argument is null or empty. Provide an argument that is not null or empty,
and then try the command again.
At C:\Users\pippo\Documents\LingangDashboard\azure-iot-remote-monitoring-mast
er\Common\Deployment\DeploymentLib.ps1:823 char:48
+     Select-AzureRmSubscription -SubscriptionId $global:SubscriptionId
+                                                ~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Set-AzureRmContext], Parameter
   BindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Microsoft.Azure
   .Commands.Profile.SetAzureRMContextCommand

Available locations:
```

**Comment by jamesweb-ms** [comment]
@robomotic I have a fix for that issue in az_national_cloud branch.  We are working on fixing with updated libraries, but you should be able to try that out soon.  I'll reply when its ready.

**Comment by robomotic** [comment]
Sure excellent.

**Comment by jamesweb-ms** [comment]
We have the final fix for this issue in [Az_national_cloud branch](https://github.com/Azure/azure-iot-remote-monitoring/tree/az_national_cloud).  We are planning to merge this branch later today after some more testing.  If you want to give it a try and give use feedback, that would be great!

**Comment by aguilaaj** [comment]
Fixed in #151 - please pull the latest from master. Closing for now.
